### PR TITLE
EVG-13419, EVG-14119: Update testresults client with new fields

### PR DESCRIPTION
**Pull request by julianedwards**
JIRA: https://jira.mongodb.org/browse/EVG-13419, https://jira.mongodb.org/browse/EVG-14119

- Revendor juniper for new protobuf files.
- Add new fields to test results client.
- Refactor options structs.

### CODE REVIEW

File: testresults/get.go [hunk 1]
```diff
@@ -11,26 +11,53 @@ import (
 	"github.com/pkg/errors"
 )
 
+type TestResultsGetOptions struct {
```
Comment (Kimchelly) [inline testresults/get.go:14]: Documentation.

File: testresults/get.go [hunk 2]
```diff
@@ -11,26 +11,53 @@ import (
...
+	Execution     int
+}
+
+func (opts *TestResultsGetOptions) validate() error {
+	if err := opts.CedarOpts.Validate(); err != nil {
+		return errors.WithStack(err)
+	}
+
+	if (opts.TaskID == "" && opts.DisplayTaskID == "") || (opts.TaskID != "" && opts.DisplayTaskID != "") {
+		return errors.New("must provide either a task id or a display task id when requesting test results")
```
Comment (Kimchelly) [inline testresults/get.go:32]: English is a bit vague about "either"; can we phrase this someting like "must provide either ...  but not both"?

File: testresults/get.go [hunk 3]
```diff
@@ -11,26 +11,53 @@ import (
...
+	// Request information. See cedar's REST documentation for more
+	// information:
+	// `https://github.com/evergreen-ci/cedar/wiki/Rest-V1-Usage`.
+	TaskID        string
+	DisplayTaskID string
+	TestName      string
+	Execution     int
+}
+
+func (opts *TestResultsGetOptions) validate() error {
```
Comment (Kimchelly) [inline testresults/get.go:26]: Should this use a catcher to accumulate and return all validation errors instead of just the first one?

**Review by Kimchelly — CHANGES_REQUESTED** [review]

File: testresults/get.go (hunk 1 above)
Comment (julianedwards) [reply testresults/get.go:16]: Done.

**Pull request merged**